cpp/src/arrow/array_dict_test.cc

```diff
@@ -486,98 +526,102 @@ TEST(TestStringDictionaryBuilder, DeltaDictionary) {
   FinishAndCheckPadding(&builder, &result_delta);
 
   // Build expected data
-  DictionaryArray expected_delta(dtype, ArrayFromJSON(int8(), "[1, 2, 1]"),
-                                 ArrayFromJSON(utf8(), "[\"test3\"]"));
+  DictionaryArray expected_delta(
+      dtype, ArrayFromJSON(int8(), "[1, 2, 1]"),
+      ArrayFromJSON(utf8(), "[\"test\", \"test2\", \"test3\"]"));
 
   ASSERT_TRUE(expected_delta.Equals(result_delta));
 }
 
-TEST(TestStringDictionaryBuilder, BigDeltaDictionary) {
-  constexpr int16_t kTestLength = 2048;
-  // Build the dictionary Array
-  StringDictionaryBuilder builder;
+// ARROW-6869(wesm): Disabled until the delta dictionary problem can be
+// addressed
 
-  StringBuilder str_builder1;
-  Int16Builder int_builder1;
+// TEST(TestStringDictionaryBuilder, BigDeltaDictionary) {
```

Comment:
Why not disable this with
```suggestion
// TEST(DISABLED_TestStringDictionaryBuilder, BigDeltaDictionary) {
```